src/builder.rs

```diff
@@ -60,6 +62,7 @@ impl UffdBuilder {
         UffdBuilder {
             close_on_exec: false,
             non_blocking: false,
+            user_mode_only: false,
```

Comment:
I would be inclined to have the default behavior be `user_mode_only: true`, so that only programs which need to be able to handle kernel-mode faults need to set an extra option. Could you make this change, and document its default value in the docstring of this method?